proposals/p5434.md

```diff
@@ -0,0 +1,1331 @@
+    -   [Use case: class accessors](#use-case-class-accessors)
+-   [Future work](#future-work)
+    -   [Temporary lifetimes](#temporary-lifetimes)
+    -   [`ref` bindings in lambdas](#ref-bindings-in-lambdas)
+    -   [Interaction with effects](#interaction-with-effects)
+    -   [More precise lifetimes](#more-precise-lifetimes)
+    -   [Combining with compile-time](#combining-with-compile-time)
+    -   [Interaction with `Call` or other interfaces](#interaction-with-call-or-other-interfaces)
+    -   [Destructuring assignment](#destructuring-assignment)
+-   [Rationale](#rationale)
+-   [Alternatives considered](#alternatives-considered)
+    -   [No `ref`, only pointers](#no-ref-only-pointers)
+    -   [Remove pointers after adding references](#remove-pointers-after-adding-references)
+    -   [Allow `ref` bindings in the fields of classes](#allow-ref-bindings-in-the-fields-of-classes)
+    -   [No call-site annotation](#no-call-site-annotation)
+    -   [Top-level `ref` introducer](#top-level-ref-introducer)
+    -   [`ref` as a type qualifier](#ref-as-a-type-qualifier)
+    -   [`bound` would change the default return to `let`](#bound-would-change-the-default-return-to-let)
+    -   [Other return conventions](#other-return-conventions)
+    -   [`return var` with compound return forms](#return-var-with-compound-return-forms)
+    -   [Other syntax for compound return forms](#other-syntax-for-compound-return-forms)
+
+<!-- tocstop -->
+
+## Abstract
+
+-   A parameter binding can be marked `ref` instead of `var` or the default. It
+    will bind to reference argument expressions in the caller and produces a
+    reference expression in the callee.
+    -   Unlike pointers, a `ref` binding can't be rebound to a different object.
+    -   This replaces `addr`, and is not restricted to the `self` parameter.
+    -   A `ref` binding, like a `let` binding, can't be used in fields of
+        classes or structs.
+    -   When calling functions, arguments to non-`self` `ref` parameters are
+        also marked with `ref`.
+-   The return of a function can optionally be marked `ref`, `let`, or `var`.
+    These control the category of the call expression invoking the function, and
+    how the return expression is returned.
+    -   These may be mixed for functions returning parens or brace forms.
+-   Any parameters whose lifetime needs to contain the lifetime of the return
+    must be marked `bound`.
+-   The address of a `ref` binding is `nocapture` and `noalias`.
+-   We mark parameters of a function that may be referenced by the return value
+    with `bound`.
+
+## Problem
+
+Reference bindings have come up multiple times:
+
+-   as a better alternative to `addr self: Self*`;
+-   for use in [lambda captures](/docs/design/lambdas.md);
+-   to support nested bindings within a destructured `var`, see
+    [issue #5250](https://github.com/carbon-language/carbon-lang/issues/5250)
```

Comment:
Lines 82 and 88 are intended to start with /docs/... but when I click them they take me to /ref/docs/... which is a 404

Not sure how to fix this without making the full links with the domain name included. It looks like it affects other links too, these are just the first two I noticed.

Reply:
Those links work if I use view mode on this file:
https://github.com/carbon-language/carbon-lang/blob/79d6ddb0b0d7357d705e35e3f42b2ccc0e897090/proposals/p5434.md

Reply:
Ah excellent! I didn't realize there was a discrepancy between the different methods of rendering a markdown document. Thanks!